src/credentials/credentialsUtilities.ts

```diff
@@ -53,3 +57,42 @@ export function notifyUserInvalidCredentials(credentialProviderId: CredentialsPr
 export function hasProfileProperty(profile: Profile, propertyName: string): boolean {
     return !!profile[propertyName]
 }
+
+/**
+ * Attempts to resolve (or refresh) a provider with a 'Cancel' progress message.
+ * User cancellation or timeout expiration will cause rejection.
+ *
+ * @param profile Profile name to display for the progress message
+ * @param provider This can be a Promise that resolves into Credentials, or void if using 'refresh'
```

Comment:
```suggestion
 * @param provider Promise that returns Credentials, or void if using 'refresh'
```

It's verbose to say "resolves into" everywhere we mean "return asynchronously". It is understood that async functions resolve promises so just "return" is clear and unambiguous (except in a context where the distinction matters).